src/api/base/index.ts

```diff
@@ -144,4 +114,41 @@ export default class BaseApi {
       body: data !== undefined ? JSON.stringify(data) : data,
     })
   }
+
+  private async fetchMultipleEnvsRecursive(
+    url: string,
+    method: 'GET' | 'POST' | 'DELETE',
+    fetchFn: BaseApi['fetch'],
+    getApiUrl: BaseApi['API_URL_GETTER'],
+    options: Omit<Options, 'env' | 'chainId'> = {},
+    envs: Env[],
+    chainId: SupportedChainId,
+    data?: unknown
+  ): Promise<Response> {
+    const { requestOptions } = options
+    // Pick the first env from the list to try
+    const [env, ...nextEnvs] = envs
+
+    const uri = getApiUrl(env)[chainId]
+
+    try {
+      const response = await fetchFn(url, method, `${uri}/${this.API_VERSION}`, data, requestOptions)
+
+      if (!response.ok && nextEnvs?.length) {
+        // If the response is not ok and there are more envs to try, do that
+        return this.fetchMultipleEnvsRecursive(url, method, fetchFn, getApiUrl, options, nextEnvs, chainId, data)
+      }
+
+      // Otherwise return response as is
+      return response
+    } catch (error) {
+      if (nextEnvs?.length) {
+        // If request failed and there are more envs to try, do that
+        return this.fetchMultipleEnvsRecursive(url, method, fetchFn, getApiUrl, options, nextEnvs, chainId, data)
+      } else {
+        // Otherwise re-throw error as is
+        throw error
```

Comment:
doesn't really matter, but i can see this will throw the error of the last env, instead of the first. Although its fine!